features/steps/dataset_feature.go

```diff
@@ -0,0 +1,240 @@
+	ctx.Step(`^the document in the database for id "([^"]*)" should be:$`, f.TheDocumentInTheDatabaseForIdShouldBe)
+}
+
+func (f *DatasetFeature) Reset() *DatasetFeature {
+	f.Datasets = make([]*models.Dataset, 0)
+	f.MongoClient.Database = memongo.RandomDatabase()
+	f.MongoClient.Init()
+	f.Config.EnablePrivateEndpoints = false
+	f.FakeAuthService.Reset()
+	return f
+}
+
+func (f *DatasetFeature) Close() error {
+	if f.svc != nil && f.ServiceRunning {
+		f.svc.Close(context.Background())
+		f.ServiceRunning = false
+	}
+	f.FakeAuthService.Close()
+	return nil
+}
+
+func (f *DatasetFeature) InitialiseService() (http.Handler, error) {
+	if err := f.svc.Run(context.Background(), "1", "", "", f.errorChan); err != nil {
+		return nil, err
+	}
+	f.ServiceRunning = true
+	return f.HTTPServer.Handler, nil
+}
+
+func funcClose(ctx context.Context) error {
+	return nil
+}
+
+func (f *DatasetFeature) DoGetHealthcheckOk(cfg *config.Configuration, buildTime string, gitCommit string, version string) (service.HealthChecker, error) {
+	return &serviceMock.HealthCheckerMock{
+		AddCheckFunc: func(name string, checker healthcheck.Checker) error { return nil },
+		StartFunc:    func(ctx context.Context) {},
+		StopFunc:     func() {},
+	}, nil
+}
+
+func (f *DatasetFeature) DoGetHTTPServer(bindAddr string, router http.Handler) service.HTTPServer {
+	f.HTTPServer.Addr = bindAddr
+	f.HTTPServer.Handler = router
+	return f.HTTPServer
+}
+
+// DoGetMongoDB returns a MongoDB
+func (f *DatasetFeature) DoGetMongoDB(ctx context.Context, cfg *config.Configuration) (store.MongoDB, error) {
+	return f.MongoClient, nil
+}
+
+func (f *DatasetFeature) DoGetGraphDBOk(ctx context.Context) (store.GraphDB, service.Closer, error) {
+	return &storeMock.GraphDBMock{CloseFunc: funcClose}, &serviceMock.CloserMock{CloseFunc: funcClose}, nil
+}
+
+func (f *DatasetFeature) DoGetKafkaProducerOk(ctx context.Context, cfg *config.Configuration) (kafka.IProducer, error) {
+	return &kafkatest.IProducerMock{
+		ChannelsFunc: func() *kafka.ProducerChannels {
+			return &kafka.ProducerChannels{}
+		},
+		CloseFunc: funcClose,
+	}, nil
+}
+
+func (f *DatasetFeature) IHaveTheseDatasets(datasetsJson *godog.DocString) error {
+
+	datasets := []models.Dataset{}
+	m := f.MongoClient
+
+	err := json.Unmarshal([]byte(datasetsJson.Content), &datasets)
+	if err != nil {
+		return err
+	}
+	s := m.Session.Copy()
+	defer s.Close()
+
+	for _, datasetDoc := range datasets {
+		f.putDatasetInDatabase(s, datasetDoc)
+	}
+
+	return nil
+}
+
+func (f *DatasetFeature) putDatasetInDatabase(s *mgo.Session, datasetDoc models.Dataset) {
+	datasetID := datasetDoc.ID
+
+	datasetUp := models.DatasetUpdate{
+		ID:      datasetID,
+		Next:    &datasetDoc,
+		Current: &datasetDoc,
+	}
+
+	update := bson.M{
+		"$set": datasetUp,
+		"$setOnInsert": bson.M{
+			"last_updated": time.Now(),
+		},
+	}
+	_, err := s.DB(f.MongoClient.Database).C("datasets").UpsertId(datasetID, update)
+	if err != nil {
+		panic(err)
```

Comment:
Better to return the err back up to the caller to this method and stop the test from running if unable to setup test data